### PR TITLE
fix: use memcached to lock daily processes on Distributed Pollers

### PR DESCRIPTION
**Pull request by murrant**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: LibreNMS/Proc.php [hunk 1]
```diff
@@ -43,6 +43,11 @@ class Proc
      */
     private $_synchronous;
 
+    /**
+     * @var int|null hold the exit code, we can only get this on the first process_status after exit
+     */
+    private $_exitcode = null;
```
Comment (laf) [inline LibreNMS/Proc.php]: Are these changes supposed to be in here? Seems totally different from the locking side.

**Comment by laf** [comment]
This now requires distributed pollers to be running php-memcached module, I can see the check for the function exists but i don't see where this falls back gracefully?

**Comment by murrant** [comment]
Hmm, that is a good point.  Thanks @laf

**Comment by murrant** [comment]
The fallback is there.  I guess I just hadn't thought of it that way.

If memcached is missing it throws a lock exception, which is caught and skips all of the code that should be inside the lock.  This is the current behavior, none of the mysql processes are run when distributed poller is enabled.

**Comment by laf** [comment]
> If memcached is missing it throws a lock exception, which is caught and skips all of the code that should be inside the lock. This is the current behavior, none of the mysql processes are run when distributed poller is enabled.

How is that current behavior? All the DB cleanup still run at the moment when you run ./daily.sh

**Comment by murrant** [comment]
It is the current behavior for distributed polling.

For non-distributed polling, it will use the existing FileLock.

**Comment by laf** [comment]
Thanks.

I'll merge this in but will exclude from the release this sunday.

**Pull request merged**